AzureMonitor/Scripts/Create-AzDiagPolicy.ps1

```diff
@@ -1414,55 +1798,155 @@ IF($($ExportEH) -or ($ExportLA))
-                $PolicyJSON = Update-LogAnalyticsJSON -resourceType $Type.ResourceType -metricsArray $metricsArray -logsArray $logsArray -nameField $RPVar[1]
-                write-host "Exporting Log Analytics Custom Azure Policy for resourceType: $($Type.ResourceType)" -ForegroundColor Yellow
-                # Make sure export directory exists!
-                if(!(Test-path "$($ExportDir)\$($RPVar[0])"))
+                $sinkDest = "LA"
+                $RPVar = Parse-ResourceType -resourceType $Type.ResourceType -sinkDest $sinkDest
+                # If we have a kind for the resourceType let's add that to the policy evaluation rules (and add it to the displayname)
+                if($Type.Kind)
                 {
-                    try
+                    $PolicyResourceDisplayName = "Apply Diagnostic Settings for $($Type.ResourceType) `($($Type.Kind)`) to a Log Analytics Workspace"
+                }
+                elseif (!($Type.Kind))
+                {
+                    $PolicyResourceDisplayName = "Apply Diagnostic Settings for $($Type.ResourceType) to a Log Analytics Workspace"
+                }
+                
+                # Create a Policy Name that is 64 chars (or less) using hash as an option [unique and repeatable]
+                $ShortNameRT = Create-Hash -StringToHash $PolicyResourceDisplayName
+
+                if($ExportInitiative)
+                {
+                    $JSONType = @'
+{
+    "type": "Microsoft.Authorization/policyDefinitions",
+    "apiVersion": "2019-09-01",
+    "name": "<SHORT NAME OF SERVICE>",
+
+'@
+                    
+                    $PolicyRSID = """[resourceId('Microsoft.Authorization/policyDefinitions/', '$($ShortNameRT)')]"""
+                    $PolicyRSIDs = $PolicyRSIDs + "                "  + $PolicyRSID + "," + "`r`n"
+                    $JSONTYPE = $JSONType.replace("<SHORT NAME OF SERVICE>", "$($ShortNameRT)")
+                    $PolicyJSON = Update-LogAnalyticsJSON -resourceType $Type.ResourceType -metricsArray $metricsArray -logsArray $logsArray -nameField $RPVar[1] -JSONType $JSONType -ExportInitiative $ExportInitiative -kind $Type.Kind -PolicyResourceDisplayName $PolicyResourceDisplayName -PolicyName $ShortNameRT
+                    $PolicyBag = $PolicyBag + $PolicyJSON[2] +  ',' + "`r`n"
+                    $PolicyDefParam = @'
                     {
-                        $NULL = new-item -ItemType Directory -Path "$($ExportDir)\$($RPVar[0])"
-                    }
-                    catch
+                        "policyDefinitionId": <PolicyResoureceID>,
+                        <Policy Definition Parameters>
+                    },
+'@
+                    $PolicyDefParam = $PolicyDefParam.Replace("<PolicyResoureceID>", "$($PolicyRSID)")
+                    $PolicyDefParam = $PolicyDefParam.Replace("<Policy Definition Parameters>", "$($PolicyJSON[3])")
```

Comment:
I'm having trouble finding where PolicyJSON[3] is instantiated and assigned.  I believe this is the root cause of my failure in attempting the following command.  If this is intended to be parameters should this be PolicyJSON[0]?
`.\Create-AzDiagPolicy.ps1 -ExportAll -ExportLA -ValidateJSON -ExportDir ".\LogPolicies3" -SubscriptionId "<subscriptionId>" -AllRegions -ExportInitiative`

Reply:
tested a little more the PolicyJSON[0] was not it... but there is a "parameters" property that is missing from the output.  This causes the Format-JSON function (a slick function btw) to fail.  I was able to test this by statically defining `'"parameters":{}'` in place of `$($PolicyJSON[3])`.

Reply:
Thank you! I'll dig into this Rob.  I appreciate it!  Stay tuned.

Reply:
I know what the issue is.  All regions missed in Log Analytics logic. That is my bad.  Resolution coming soon.  Thank you!

Reply:
This is fixed with the latest commit, please review Rob @rwsanders

Reply:
great it exports the templates and the templates deploy.  thanks!